lua/ccc/ui.lua

```diff
@@ -322,15 +326,25 @@ function UI:highlight()
     for j = 1, bar_len do
       end_ = update_end(j == point_idx, start, #bar_char, #point_char)
 
-      local hex
-      if point_color ~= "" and j == point_idx then
-        hex = point_color
+      local new_value = (j - 0.5) / bar_len * (max - min) + min
+      local hex = self.color:hex(i, new_value)
+      local hi = {}
+      if j == point_idx then
+        hi.fg = hex
+        if not empty_point_bg then
+          local RGB = self.color:get_rgb()
+          local _, _, L = unpack(convert.rgb2hsl(RGB))
+          hi.fg = L < 0.6 and point_color_on_dark or point_color_on_light
```

Comment:
You can use this to calculate brightness. Right now it's a local function, but you can make it public if you want. You would need to modify it a bit, though, as you want to calculate from RGB values, not from strings.
https://github.com/uga-rosa/ccc.nvim/blob/78aed26e18a087501be0475443b5a623adbf6290/lua/ccc/utils/init.lua#L117-L126

Reply:
I see. Let me give it a try.